### PR TITLE
Add campaign burndown chart

### PR DESCRIPTION
**Pull request by felixfbecker**
Blocked by https://github.com/sourcegraph/sourcegraph/issues/6011

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -19,7 +19,6 @@
     "all:typecheck": "gulp generate && dev/foreach-ts-project.sh yarn -s run typecheck -p tsconfig.json",
     "graphql": "gulp graphQLTypes",
     "graphql-lint": "graphql-schema-linter --old-implements-syntax --comment-descriptions cmd/frontend/graphqlbackend/schema.graphql",
-    "prepublish": "gulp generate",
```
Comment (lguychard) [inline package.json]: Is this expected? Looks like there's more unintentional changes in Procfile, generate.sh and handle-change.sh

Comment (felixfbecker) [reply package.json]: Yep, sorry, this was not ready for review yet.

**Comment by tsenart** [comment]
By merging https://github.com/sourcegraph/sourcegraph/pull/5915, this PR has now conflicts that need to be resolved. Marking as release blocker.

File: web/src/enterprise/campaigns/detail/BurndownChart.tsx [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+import { ICampaign } from '../../../../../shared/src/graphql/schema'
+
+interface Props extends Pick<ICampaign, 'changesetCountsOverTime'> {
+    history: H.History
+}
+
+const dateTickFormat = new Intl.DateTimeFormat(undefined, { month: 'long', day: 'numeric' })
+const dateTickFormatter = (timestamp: number): string => dateTickFormat.format(timestamp)
+
+// const tooltipLabelFormat = new Intl.DateTimeFormat(undefined, { dateStyle: 'medium' })
```
Comment (lguychard) [inline web/src/enterprise/campaigns/detail/BurndownChart.tsx]: Leftover comment?

File: web/src/enterprise/campaigns/detail/CampaignDetails.tsx [hunk 3]
```diff
@@ -82,6 +72,11 @@ export const CampaignDetails: React.FunctionComponent<Props> = ({ campaignID, hi
         return () => subscription.unsubscribe()
     }, [campaignID, triggerError])
 
+    const queryChangesetsConnection = useCallback(
+        (args: FilteredConnectionQueryArgs) => queryChangesets(campaignID!, args),
```
Comment (lguychard) [inline web/src/enterprise/campaigns/detail/CampaignDetails.tsx]: I see what you mean about the annoyance of having to call this at the top-level, this is frustrating 😞

File: web/src/enterprise/campaigns/detail/CampaignDetails.tsx [hunk 4]
```diff
@@ -65,7 +55,7 @@ export const CampaignDetails: React.FunctionComponent<Props> = ({ campaignID, hi
 
     useEffect(() => {
         if (campaignID) {
-            // The namespace of a campaign can only be set on creation.
+            // Namespace cannot be edited
```
Comment (lguychard) [inline web/src/enterprise/campaigns/detail/CampaignDetails.tsx]:
```suggestion
            // The namespace of a campaign can only be set on creation.
```

**Comment by tsenart** [comment]
Merging this to tag `3.9.0-rc.2`

File: web/src/enterprise/campaigns/detail/BurndownChart.tsx (hunk 2 above)
Comment (felixfbecker) [reply web/src/enterprise/campaigns/detail/BurndownChart.tsx]: `dateStyle: 'medium'` is supported in Chrome but not in lib.d.ts yet

File: web/src/enterprise/campaigns/detail/backend.ts [hunk 5]
```diff
@@ -38,6 +38,21 @@ const campaignFragment = gql`
             openApproved
             openChangesRequested
             openPending
+            nodes {
```
Comment (felixfbecker) [inline web/src/enterprise/campaigns/detail/backend.ts]: @hadrian-git FYI this was a wrong merge - `ChangesetCounts` does not have `nodes`

image

**Pull request merged**